src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -88,30 +93,47 @@ internal class CommandUsingCommandMessageWireProtocol<TCommandResult> : IWirePro
             }
         }
 
+        // public properties
+        public bool MoreToCome => _moreToCome;
+
         // public methods
         public TCommandResult Execute(IConnection connection, CancellationToken cancellationToken)
         {
             try
             {
-                var message = CreateCommandMessage(connection.Description);
-                message = AutoEncryptFieldsIfNecessary(message, connection, cancellationToken);
-
-                try
+                bool responseExpected;
+                int responseTo;
+                if (_moreToCome)
                 {
-                    connection.SendMessage(message, _messageEncoderSettings, cancellationToken);
+                    responseExpected = true;
+                    responseTo = _previousRequestId;
                 }
-                finally
+                else
                 {
-                    if (message.WasSent)
+                    var message = CreateCommandMessage(connection.Description);
```

Comment:
https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/136/files#r433318568

Reply:
Suggested code:

	public TCommandResult Execute(IConnection connection, CancellationToken cancellationToken)
	{
		try
		{
			bool responseExpected;
			int responseTo;
			if (_moreToCome)
			{
				responseExpected = true;
				responseTo = _previousRequestId;
			}
			else
			{
				var message = CreateCommandMessage(connection.Description);
				message = AutoEncryptFieldsIfNecessary(message, connection, cancellationToken);

				try
				{
					connection.SendMessage(message, _messageEncoderSettings, cancellationToken);
				}
				finally
				{
					if (message.WasSent)
					{
						MessageWasProbablySent(message);
					}
				}

				responseExpected = _responseHandling != CommandResponseHandling.NoResponseExpected;
				responseTo = message.RequestId;
			}

			if (responseExpected)
			{
				var encoderSelector = new CommandResponseMessageEncoderSelector();
				var response = (CommandResponseMessage)connection.ReceiveMessage(responseTo, encoderSelector, _messageEncoderSettings, cancellationToken);
				SaveResponseInfoIfRequired(response); // why not just save always?
				response = AutoDecryptFieldsIfNecessary(response, cancellationToken);
				return ProcessResponse(connection.ConnectionId, response.WrappedMessage);
			}
			else
			{
				return default(TCommandResult);
			}
		}
		catch (Exception exception)
		{
			AddErrorLabelIfRequired(exception, connection.Description.ServerVersion);

			TransactionHelper.UnpinServerIfNeededOnCommandException(_session, exception);
			throw;
		}
	}

Reply:
done but the change in this line:

    responseExpected = _responseHandling != CommandResponseHandling.NoResponseExpected;

since we can change responseExpected value in the encoders level. Now it's:

     responseExpected = message.WrappedMessage.ResponseExpected;